### PR TITLE
Refactored Router to be a non-static class

### PR DESCRIPTION
**Pull request by CommandString**
# Why?
In attempts to make all of the libraries in TNAPF comply with PSR and be usable in any circumstance. It only makes sense to convert Router into a non-static class so it can be used with PSR-15. Which will be implemented soon

### CODE REVIEW

File: src/Routing/Route.php [hunk 1]
```diff
@@ -30,28 +30,28 @@ class Route
      * @param class-string<RequestHandlerInterface> $controller
      * @param Methods                               ...$methods
      */
```
Comment (xHeaven) [inline src/Routing/Route.php:32]: the phpdoc is incomplete here

File: src/Routing/Route.php [hunk 2]
```diff
@@ -30,28 +30,28 @@ class Route
...
-    public function __construct(string $uri, public readonly string $controller, Methods ...$methods)
+    public function __construct(protected readonly Router $router, string $uri, public readonly string $controller, Methods ...$methods)
```
Comment (xHeaven) [inline src/Routing/Route.php:33]: you should consider breaking down the arguments into newlines to comply with PSR12
eg:
```php
public function __construct(
        protected readonly Router $router,
        string $uri,
        public readonly string $controller,
        Methods ...$methods
    ) {
        // ...
```

File: tests/RouteTests.php [hunk 3]
```diff
@@ -5,13 +5,23 @@
...
+        $this->router = new Router();
+    }
+
     public function createBasicRoute(Methods...$methods): Route
     {
-        return new Route("home", TestController::class, ...$methods);
+        return new Route($this->router, "home", TestController::class, ...$methods);
     }
 
     public function testRoutePrependsMissingStartingSlash(): void
```
Comment (xHeaven) [inline tests/RouteTests.php:27]: use `assertSame` instead of `assertEquals` here, as `assertSame` is typesafe

File: tests/RouteTests.php [hunk 4]
```diff
@@ -24,7 +34,7 @@ public function testRouteRejectsInvalidController(): void
     {
         $this->expectException(InvalidArgumentException::class);
 
-        new Route("/", "InvalidController");
+        new Route($this->router, "/", "InvalidController");
     }
 
     public function testRouteSettingGettingParameters(): void
```
Comment (xHeaven) [inline tests/RouteTests.php:40]: use `assertSame` instead of `assertEquals` here, as `assertSame` is typesafe

**Review by xHeaven — CHANGES_REQUESTED** [review]

**Review by xHeaven — CHANGES_REQUESTED** [review]
The changed files are looking good now, but please do a commit with the updated examples as well. Currently, you are calling non-static methods statically in your examples.

**Comment by CommandString** [comment]
> The changed files are looking good now, but please do a commit with the updated examples as well. Currently, you are calling non-static methods statically in your examples. Update the README too.

Examples are getting nuked and there's gonna be more changes which is why the README hasn't been updated

**Review by xHeaven — APPROVED** [review]
Alright then, sounds good. Approved.

**Pull request merged**